website/components/AppWrapper.tsx

```diff
@@ -61,17 +61,25 @@ export default function AppWrapper(inputProps: AppWrapperProps): ReactElement {
   const location = useLocation();
   const navigation = useNavigate();
 
-  const [viewerSettings, setViewerSettings] = useState<Partial<GlobalViewerSettings>>(props.viewerSettings);
-  const [viewerArgs, setViewerArgs] = useState<AppDataProps>(props.viewerArgs);
+  const [viewerSettings, setViewerSettings] = useState<Partial<GlobalViewerSettings> | null>(null);
+  const [viewerArgs, setViewerArgs] = useState<AppDataProps | null>(null);
   const [searchParams] = useSearchParams();
 
   useEffect(() => {
     // On load, fetch parameters from the URL and location state, then merge.
     const locationArgs = location.state as AppDataProps;
-    getArgsFromParams(searchParams).then(({ args: urlArgs, viewerSettings: urlViewerSettings }) => {
-      setViewerArgs({ ...DEFAULT_APP_PROPS, ...urlArgs, ...locationArgs });
-      setViewerSettings({ ...DEFAULT_VIEWER_SETTINGS, ...urlViewerSettings });
-    });
+    getArgsFromParams(searchParams).then(
+      ({ args: urlArgs, viewerSettings: urlViewerSettings }) => {
+        console.log("URL viewer settings", urlViewerSettings);
+        setViewerArgs({ ...DEFAULT_APP_PROPS, ...props.viewerArgs, ...urlArgs, ...locationArgs });
+        setViewerSettings({ ...DEFAULT_VIEWER_SETTINGS, ...props.viewerSettings, ...urlViewerSettings });
+      },
+      (reason) => {
+        console.warn("Failed to parse URL parameters: ", reason);
+        setViewerArgs({ ...DEFAULT_APP_PROPS, ...props.viewerArgs, ...locationArgs });
+        setViewerSettings({ ...DEFAULT_VIEWER_SETTINGS, ...props.viewerSettings });
+      }
+    );
```

Comment:
Second part here triggers if the `getArgsFromParams` operation fails. That way the viewer args + settings will always be initialized.